controllers/argocd/deployment.go

```diff
@@ -965,6 +965,21 @@ func (r *ReconcileArgoCD) reconcileRepoDeployment(cr *argoproj.ArgoCD, useTLSFor
 		repoEnv = argoutil.EnvMerge(repoEnv, []corev1.EnvVar{{Name: "ARGOCD_EXEC_TIMEOUT", Value: fmt.Sprintf("%ds", *cr.Spec.Repo.ExecTimeout)}}, true)
 	}
 
+	// if running in a FIPS enabled host, set the GODEBUG env wit appropriate value
+	fipsConfigChecker := r.FipsConfigChecker
+	if fipsConfigChecker != nil {
+		fipsEnabled, err := fipsConfigChecker.IsFipsEnabled()
+		if err != nil {
+			return err
+		}
```

Comment:
Do we want to fail on error or just log and continue? On non-FIPS clusters this might be disruptive if restricted policies prevent reading `/proc/sys/crypto/fips_enabled`. It’s a rare case but just highlighting.

Reply:
if file is not existent, we don't return an error. We check for non existent file path and ignore the error.  In the `fileExists` function, I have put this logic.
```
if err != nil {
    if os.IsNotExist(err) {
	    return false, nil
    }
    return false, err
}
```

Reply:
Okay. And are permission issues handled in similar way?